src/lib.rs

```diff
@@ -1,3 +1,162 @@
-#[test]
-fn it_works() {
+#[cfg(test)]
+mod test {
+    use tokens::{ Lexer, LexerOptions };
+
+    #[test]
+    fn lexer_usage() {
+        // build the lexer once for project environment with extensions.
+        let lexer = Lexer::new(LexerOptions::default(), vec![]);
+
+        // use many times.
+        {
+            let source: String = "{{ var }}".into();
+            for token in lexer.tokens(&source) {
+                println!("{:?}", token);
+            }
+        }
+    }
+}
+
+// I don't know where to put this, keeping it in root for now.
+#[derive(Debug, Default, Clone)]
+pub struct Position {
```

Comment:
In my design, the lexer has a helper object called "cursor" its supposed to keep track of the current position and do conversions like char-position to line:col. Maybe somewhere around there?

Actually I didn't use a struct like this, because everytime I need to output a position is an error. I then do something like `err!(CODE, "something happened at {pos}", pos = cursor);

```
impl<'a> fmt::Display for Cursor<'a> {
    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
        write!(f, "`{template_name}` line {line} column {column}",
            template_name = self.template.name(),
            line = self.line(),
            column = self.column())
    }
}
```